src/librbd/internal.cc

```diff
@@ -2107,6 +2107,15 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
         return -EBUSY;
       }
 
+      std::map<std::string, bufferlist> out_vals;
+      r = io_ctx.omap_get_vals(header_oid, "", RBD_GROUP_UPLINK, 1, &out_vals);
+      if (out_vals.size() > 0) {
+	lderr(cct) << "image is in a consistency group - not removing" << dendl;
+	ictx->owner_lock.put_read();
+	ictx->state->close();
+	return -EMLINK;
```

Comment:
... never mind, I see that you are using it for nicer error output from the rbd CLI.